corehq/apps/data_analytics/malt_generator.py

```diff
@@ -0,0 +1,74 @@
+from corehq.apps.app_manager.models import Application
+from corehq.apps.data_analytics.models import MALTRow
+from corehq.apps.domain.models import Domain
+from corehq.apps.smsforms.app import COMMCONNECT_DEVICE_ID
+from corehq.apps.sofabed.models import FormData
+
+
+class MALTTableGenerator(object):
+    """
+        Populates SQL table with data for given datespan
+        See .models.MALTRow
+    """
+    _wam_pam_cache = {}
+
+    def __init__(self, datespan_object):
+        self.datespan = datespan_object
+
+    def build_table(self):
+
+        for domain in Domain.get_all():
+            malt_rows_to_save = []
+            for user in domain.all_users():
+                forms_query = self.get_forms_queryset(user._id, domain.name)
+                num_of_forms = forms_query.count()
+                apps_submitted_for = [app_id for (app_id,) in
+                                      forms_query.values_list('app_id').distinct()]
+
+                for app_id in apps_submitted_for:
+                    wam, pam = self._wam_pams(domain.name, app_id)
+                    db_row = MALTRow(
+                        month=self.datespan.startdate,
+                        user_id=user._id,
+                        username=user.username,
+                        email=user.email,
+                        is_web_user=user.doc_type == 'WebUser',
+                        domain_name=domain.name,
+                        num_of_forms=num_of_forms,
+                        app_id=app_id,
+                        wam=wam,
+                        pam=pam
+                    )
+                    malt_rows_to_save.append(db_row)
+            MALTRow.objects.bulk_create(malt_rows_to_save)
```

Comment:
If we want to make this idempotent (which I think we do) then this should be more of an upsert than a create. otherwise if you ran it twice you would have multiple rows per user/month/domain which is wrong. adding the db constraint I mentioned below will also safeguard against that